test/blackbox-tests/test-cases/deprecated-library-name/name-clash.t

```diff
@@ -0,0 +1,20 @@
+The old_public_name field is named in a confusing matter. As we can see in this
+test, the name may refer to private names in the current scope as well as public
+names.
```

Comment:
The explanation doesn't seem correct to me. `(old_public_name b)` doesn't refer to an existing name, it defines a new name `b` as a redirection to another one. You'd have the same error message if you replaced `(deprecated_library_name ...)` by `(library (public_name b))`.

In this example when Dune tries to understand what `b` means it has two choices:

- `b` can be interpreted as one of the deprecated name of library `a`
- `b` can be interpreted as the private library `b`

both choices are valid, so Dune can't make a clear decision.

Reply:
I suppose this test doesn't really clarify anything then. I'm removing it.